1-Draft/RFC00XX-Native-Command-Error-Handling.md

```diff
@@ -0,0 +1,229 @@
+---
+RFC: RFC00XX
+Author: Jason Helmick
+Status: Draft
+Area: Core
+Comments Due: 10/31/2020
+---
+
+# Native Command Error Handling
+
+PowerShell scripts using native commands would benefit from being able to use error handling
+features like those used by cmdlets.
+
+## Motivation
+
+In PowerShell by default, script processing continues when non-terminating errors occur. This is a
+benefit when expecting non-terminating errors in normal execution such as non-responsive computers from
+a list. This default behavior is controlled with the preference variable
+`$ErrorActionPreference` default of `Continue`.
+
+During development and debugging, often customers prefer that script execution stops when a
+non-terminating error occurs. PowerShell currently supports customers with this ability by setting
+the preference variable in the script.
+
+```Powershell
+$ErrorActionPreference = 'Stop'
+```
+
+Native commands return an exit code to the calling application which will be zero for success or
+non-zero for failure. However, native commands currently do not participate in the PowerShell error
+stream. Customers working with native commands in their scripts will need to check the execution
+status after each call using a helper function similar to below:
+
+```Powershell
+if( ! $? )
+{
+    exit $lastexitcode;
+}
+ ```
+
+Simply relaying the errors through the error stream isn't the solution. The example itself doesn't
+support all cases as `$?` can be false from a cmdlet or function error, making `$LASTEXITCODE`
+stale.
```

Comment:
this might also argue for improving `$LASTEXITCODE` behavior so is less or not stale